airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -423,12 +425,37 @@ def create_declarative_stream(self, model: DeclarativeStreamModel, config: Confi
             primary_key=model.primary_key,
             retriever=retriever,
             schema_loader=schema_loader,
-            stream_cursor_field=model.stream_cursor_field or [],
+            stream_cursor_field=cursor_field or [],
             transformations=transformations,
             config=config,
             parameters={},
         )
 
+    @staticmethod
+    def _get_cursor_field_from_stream_slicer(model: DeclarativeStreamModel) -> Optional[Union[str, List[str]]]:
+        if not model.retriever or not model.retriever.stream_slicer:
+            return None
+        elif hasattr(model.retriever.stream_slicer, "cursor_field"):
+            return model.retriever.stream_slicer.cursor_field
+        elif isinstance(model.retriever.stream_slicer, CartesianProductStreamSlicerModel):
```

Comment:
This is where things get just a little bit crazy specifically for Posthog and Zenloop because they use either a CartesianProductStreamSlicer or a custom CartesianProductStreamSlicer so we cannot easily derive what the `cursor_field` should be since there are multiple underlying slicers.

This is a temporary hack where we look for the first slicer we can find that has a cursor_field and use that for the current stream. This isn't a long term solution as the comments mention. Rather, once we separate the `incremental` and `iterable` concepts we can then directly reference the incremental implementation's cursor field and get rid of the existing conditions here.

Reply:
Death to this method (... eventually)!